rpcserverhandlers_test.go

```diff
@@ -6,28 +6,854 @@ package main
+			}},
+			Amount: map[string]int64{"DcuQKx8BES9wU7C6Q5VmLBjw436r27hayjS": -1},
+			COuts: []types.SStxCommitOut{{
+				Addr:       "DsRah84zx6jdA4nMYboMfLERA5V3KhBr4ru",
+				CommitAmt:  100000000,
+				ChangeAddr: "DsfkbtrSUr5cFdQYq3WSKo9vvFs5qxZXbgF",
+				ChangeAmt:  0,
+			}},
+		},
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInvalidParameter,
+	}, {
+		name:    "handleCreateRawSStx: invalid address",
+		handler: handleCreateRawSStx,
+		cmd: &types.CreateRawSStxCmd{
+			Inputs: []types.SStxInput{{
+				Txid: "e02f03a25a57afdd402818fe5b13985a0731502ad8a8c93d1874900e84d3330d",
+				Vout: 0,
+				Tree: 0,
+				Amt:  100000000,
+			}},
+			Amount: map[string]int64{"DcuqInvalidwU7C6Q5VmLBjw436r27hayjS": 100000000},
+			COuts: []types.SStxCommitOut{{
+				Addr:       "DsRah84zx6jdA4nMYboMfLERA5V3KhBr4ru",
+				CommitAmt:  100000000,
+				ChangeAddr: "DsfkbtrSUr5cFdQYq3WSKo9vvFs5qxZXbgF",
+				ChangeAmt:  0,
+			}},
+		},
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInvalidAddressOrKey,
+	}, {
+		name:    "handleCreateRawSStx: invalid address type",
+		handler: handleCreateRawSStx,
+		cmd: &types.CreateRawSStxCmd{
+			Inputs: []types.SStxInput{{
+				Txid: "e02f03a25a57afdd402818fe5b13985a0731502ad8a8c93d1874900e84d3330d",
+				Vout: 0,
+				Tree: 0,
+				Amt:  100000000,
+			}},
+			Amount: map[string]int64{"DkM3EyZ546GghVSkvzb6J47PvGDyntqiDtFgipQhNj78Xm2mUYRpf": 100000000},
+			COuts: []types.SStxCommitOut{{
+				Addr:       "DsRah84zx6jdA4nMYboMfLERA5V3KhBr4ru",
+				CommitAmt:  100000000,
+				ChangeAddr: "DsfkbtrSUr5cFdQYq3WSKo9vvFs5qxZXbgF",
+				ChangeAmt:  0,
+			}},
+		},
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInvalidAddressOrKey,
+	}, {
+		name:    "handleCreateRawSStx: unsupported dsa",
+		handler: handleCreateRawSStx,
+		cmd: &types.CreateRawSStxCmd{
+			Inputs: []types.SStxInput{{
+				Txid: "e02f03a25a57afdd402818fe5b13985a0731502ad8a8c93d1874900e84d3330d",
+				Vout: 0,
+				Tree: 0,
+				Amt:  100000000,
+			}},
+			Amount: map[string]int64{"DSXcZv4oSRiEoWL2a9aD8sgfptRo1YEXNKj": 100000000},
+			COuts: []types.SStxCommitOut{{
+				Addr:       "DsRah84zx6jdA4nMYboMfLERA5V3KhBr4ru",
+				CommitAmt:  100000000,
+				ChangeAddr: "DsfkbtrSUr5cFdQYq3WSKo9vvFs5qxZXbgF",
+				ChangeAmt:  0,
+			}},
+		},
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInternal.Code,
+	}, {
+		name:    "handleCreateRawSStx: change amount greater than input amount",
+		handler: handleCreateRawSStx,
+		cmd: &types.CreateRawSStxCmd{
+			Inputs: []types.SStxInput{{
+				Txid: "e02f03a25a57afdd402818fe5b13985a0731502ad8a8c93d1874900e84d3330d",
+				Vout: 0,
+				Tree: 0,
+				Amt:  100000000,
+			}},
+			Amount: map[string]int64{"DcuQKx8BES9wU7C6Q5VmLBjw436r27hayjS": 100000000},
+			COuts: []types.SStxCommitOut{{
+				Addr:       "DsRah84zx6jdA4nMYboMfLERA5V3KhBr4ru",
+				CommitAmt:  100000000,
+				ChangeAddr: "DsfkbtrSUr5cFdQYq3WSKo9vvFs5qxZXbgF",
+				ChangeAmt:  200000000,
+			}},
+		},
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInvalidParameter,
+	}, {
```

Comment:
Regarding this comment and a couple of others below - yes, there are a few error scenarios that are unreachable, primarily due to the given rpcserver handler already checking for the same error scenario(s) that are handled in the method that it is calling. If we still want to test it, we could stub the method being called to return an error in these cases (since it can't be hit with just bad input parameters).

Since these are unreachable, we could ignore errors returned by the method being called in these cases, but that would make this code more coupled to the internals of that method (i.e., if a new error case was added, then the rpcserver code would need to be updated).